lib/vehicle_factory.rb

```diff
@@ -0,0 +1,30 @@
+class VehicleFactory 
+  attr_accessor :registration_date
+    attr_reader :engine,
+                :make,
+                :model,
+                :plate_type,
+                :vin,
+                :year,
+                :electric_vehicle_type,
+                :vin_1_10,
+                :dol_vehicle_id,
+                :model_year,
+                :vehicles
+  def initialize 
+    @vehicles = []
+  end 
+
+  def create_vehicles(details)
+    #vehicles = []
+    details.map do |vehicle_data|
+      Vehicle.new({
+        vin: vehicle_data[:vin_1_10],
+        year: vehicle_data[:model_year],
+        make: vehicle_data[:make],
+        model: vehicle_data[:model],
+        engine: :ev
+      })
+    end
+  end
+end
```

Comment:
You do not need any of these attr_accessors or attr_readers. The interaction pattern mentioned that you just need a `create_vehicles` method that returns a collection of vehicle objects. We don't need any other methods to return something else in this class. So, we can slim this class up to look something like: 

```ruby 
class VehicleFactory 

  def create_vehicles(details)
    details.map do |vehicle_data|
      Vehicle.new({
        vin: vehicle_data[:vin_1_10],
        year: vehicle_data[:model_year],
        make: vehicle_data[:make],
        model: vehicle_data[:model],
        engine: :ev
      })
    end
  end
end
```